pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/stats/MetadataStoreStats.java

```diff
@@ -0,0 +1,109 @@
+    private static final String OPS_TYPE_LABEL_NAME = "type";
+    private static final String METADATA_STORE_LABEL_NAME = "name";
+    private static final String OPS_TYPE_GET = "get";
+    private static final String OPS_TYPE_DEL = "del";
+    private static final String OPS_TYPE_PUT = "put";
+    protected static final String PREFIX = "pulsar_metadata_store_";
+
+    private static final Histogram OPS_SUCCEED = Histogram
+            .build(PREFIX + "ops_latency", "-")
+            .unit("ms")
+            .buckets(BUCKETS)
+            .labelNames(METADATA_STORE_LABEL_NAME, OPS_TYPE_LABEL_NAME)
+            .register();
+    private static final Counter OPS_FAILED = Counter
+            .build(PREFIX + "ops_failed", "-")
+            .labelNames(METADATA_STORE_LABEL_NAME, OPS_TYPE_LABEL_NAME)
+            .register();
+    private static final Counter PUT_BYTES = Counter
+            .build(PREFIX + "put_", "-")
+            .unit("bytes")
+            .labelNames(METADATA_STORE_LABEL_NAME)
+            .register();
+
+    private final Histogram.Child getOpsSucceedChild;
+    private final Histogram.Child delOpsSucceedChild;
+    private final Histogram.Child putOpsSucceedChild;
+    private final Counter.Child getOpsFailedChild;
+    private final Counter.Child delOpsFailedChild;
+    private final Counter.Child putOpsFailedChild;
+    private final Counter.Child putBytesChild;
+    private final String metadataStoreName;
+    private final AtomicBoolean closed = new AtomicBoolean(false);
+
+    public MetadataStoreStats(String metadataStoreName) {
+        this.metadataStoreName = metadataStoreName;
+
+        this.getOpsSucceedChild = OPS_SUCCEED.labels(metadataStoreName, OPS_TYPE_GET);
+        this.delOpsSucceedChild = OPS_SUCCEED.labels(metadataStoreName, OPS_TYPE_DEL);
+        this.putOpsSucceedChild = OPS_SUCCEED.labels(metadataStoreName, OPS_TYPE_PUT);
+        this.getOpsFailedChild = OPS_FAILED.labels(metadataStoreName, OPS_TYPE_GET);
+        this.delOpsFailedChild = OPS_FAILED.labels(metadataStoreName, OPS_TYPE_DEL);
+        this.putOpsFailedChild = OPS_FAILED.labels(metadataStoreName, OPS_TYPE_PUT);
+        this.putBytesChild = PUT_BYTES.labels(metadataStoreName);
+    }
+
+    public void recordGetOpsLatency(long millis) {
+        this.getOpsSucceedChild.observe(millis);
+    }
+
+    public void recordDelOpsLatency(long millis) {
+        this.delOpsSucceedChild.observe(millis);
+    }
+
+    public void recordPutOpsLatency(long millis, int bytes) {
+        this.putOpsSucceedChild.observe(millis);
+        this.putBytesChild.inc(bytes);
+    }
+
+    public void recordGetOpsFailed() {
+        this.getOpsFailedChild.inc();
+    }
+
+    public void recordDelOpsFailed() {
+        this.delOpsFailedChild.inc();
+    }
+
+    public void recordPutOpsFailed() {
+        this.putOpsFailedChild.inc();
+    }
+
+    @Override
+    public void close() throws Exception {
+        if (this.closed.compareAndSet(false, true)) {
```

Comment:
I wouldn't bother with this check - it's a developer bug to call close twice

Reply:
of cause it's unnecessary in non-test code, but according to my experience, everything happens in tests, this is to prevent ci tests failure

Reply:
Then let us find the bug in test when it happens: Throw an exception if we called close, and it's already closed.

Reply:
There are many similar uses in Pulsar, would you change them all?

Reply:
If you throw `RuntimeException`, you are not required to change, no? Or do you mean, there are many tests that call close() twice, which will cause your build to fail on many tests?

Reply:
> there are many tests that call close() twice, which will cause your build to fail on many tests?

right, when I develop offloader metrics, it happened, so I have to add the logic